src/main/60-eks.tf

```diff
@@ -70,7 +77,9 @@ module "eks" {
 
   fargate_profile_defaults = {
     iam_role_additional_policies = {
-      fargate_logging = aws_iam_policy.fargate_profile_logging.arn
+      fargate_logging         = aws_iam_policy.fargate_profile_logging.arn
+      cloudwatch_agent_server = data.aws_iam_policy.cloudwatch_agent_server.arn
+      xray_daemon_write       = data.aws_iam_policy.xray_daemon_write.arn
```

Comment:
Are you sure the Fargate profile role needs these permissions if you already gave them to the pod's serviceaccount?